pkg/handler/project/project.go

```diff
@@ -307,3 +308,67 @@ func (h *handler) GetStaffs(c *gin.Context) {
 	c.JSON(http.StatusOK, view.CreateResponse(view.ToProjectMemberListData(members, heads),
 		&view.PaginationResponse{Pagination: query.Pagination, Total: total}, nil, nil))
 }
+
+// DeleteMember godoc
+// @Summary Delete member in a project
+// @Description Delete member in a project
+// @Tags Project
+// @Accept  json
+// @Produce  json
+// @Param Authorization header string true "jwt token"
+// @Param id path string false "member ID"
+// @Success 200 {object} view.DeleteMemberMessageResponse
+// @Failure 400 {object} view.ErrorResponse
+// @Failure 404 {object} view.ErrorResponse
+// @Failure 500 {object} view.ErrorResponse
+// @Router /projects/members/:id [delete]
+func (h *handler) DeleteMember(c *gin.Context) {
+	memberID := c.Param("id")
+
+	// TODO: can we move this to middleware ?
+	l := h.logger.Fields(logger.Fields{
+		"handler":   "project",
+		"method":    "DeleteMember",
+		"projectID": memberID,
+	})
+
+	// get member infor
+	projectMember, err := h.store.ProjectMember.One(h.repo.DB(), memberID)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			l.Error(err, "project member not found")
+			c.JSON(http.StatusNotFound, view.CreateResponse[any](nil, nil, ErrProjectMemberNotFound, memberID))
+			return
+		}
+		l.Error(err, "error query member from db")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, memberID))
+		return
+	}
+
+	if projectMember.Status == model.ProjectMemberStatusInactive {
+		l.Error(ErrInactiveMember, "error inactive member")
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, ErrInactiveMember, memberID))
+		return
+	}
+
+	// Begin transaction
+	tx, done := h.repo.NewTransaction()
+
+	slotID := projectMember.ProjectSlotID
+
+	err = h.store.ProjectMember.HardDelete(tx.DB(), memberID)
+	if err != nil {
+		l.Error(err, "error delete project member")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), memberID))
+		return
+	}
+
+	err = h.store.ProjectSlot.HardDelete(tx.DB(), slotID.String())
+	if err != nil {
+		l.Error(err, "error delete project member")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, done(err), memberID))
+		return
+	}
+
+	c.JSON(http.StatusOK, view.CreateResponse[any](view.ToDeleteMemberMessage("successed"), nil, done(nil), nil))
```

Comment:
should implement the general response for these apis

Reply:
updated